sunscreen/src/types/zkp/program_node.rs

```diff
@@ -163,6 +160,22 @@ where
     }
 }
 
+impl<T> CreateSharedZkpProgramInput for ProgramNode<T>
+where
+    T: SharedZkpType,
+{
+    fn shared_input(plaintext_modulus: u64) -> Self {
```

Comment:
Should we add a note that this had to be in a certain order? I guess I am a tad concerned that this is called before/after adding the private inputs and then the `ctx.add_private_input` line adds shared input in the wrong order.

Reply:
There's no way for the user to add them out of order if that's what you mean. I can add a comment to the one place where inputs get combined in `sunscreen_runtime::linked` to document the internal requirement.